application-load-balancer/main.tf

```diff
@@ -166,3 +174,267 @@ output "cert-arn" {
+
+  name        = "${var.application}-${var.environment}-ACL"
+  description = "CloudFront Origin Verify"
+  scope       = "REGIONAL"
+
+  default_action {
+    block {} # Action to perform if none of the rules contained in the WebACL match
+  }
+
+  visibility_config {
+    cloudwatch_metrics_enabled = true
+    metric_name                = "${var.application}-${var.environment}-XOriginVerify"
+    sampled_requests_enabled   = true
+  }
+
+  rule {
+    name     = "${var.application}-${var.environment}-XOriginVerify"
+    priority = "0"
+
+    action {
+      allow {}
+    }
+
+    visibility_config {
+      cloudwatch_metrics_enabled = true
+      metric_name                = "${var.application}-${var.environment}-XMetric"
+      sampled_requests_enabled   = true
+    }
+
+    statement {
+      or_statement {
+        statement {
+          byte_match_statement {
+            field_to_match {
+              single_header {
+                name = "x-origin-verify"
+              }
+            }
+            positional_constraint = "EXACTLY"
+            search_string         = jsondecode(data.aws_secretsmanager_secret_version.origin_verify_secret_version.secret_string)["HEADERVALUE"]
+            text_transformation {
+              priority = 0
+              type     = "NONE"
+            }
+          }
+        }
+        statement {
+          byte_match_statement {
+            field_to_match {
+              single_header {
+                name = "x-origin-verify"
+              }
+            }
+            positional_constraint = "EXACTLY"
+            search_string         = random_password.origin-secret.result
+            text_transformation {
+              priority = 0
+              type     = "NONE"
+            }
+          }
+        }
+      }
+    }
+  }
+
+  # Add the Managed Rule Group for Log4j2 protection
+  rule {
+    name     = "AWS-AWSManagedRulesKnownBadInputsRuleSet"
+    priority = 1
+
+    override_action {
+      none {}
+    }
+
+    visibility_config {
+      cloudwatch_metrics_enabled = true
+      metric_name                = "${var.application}-${var.environment}-WAF-ACL-ManagedBadInputs"
+      sampled_requests_enabled   = true
+    }
+
+    statement {
+      managed_rule_group_statement {
+        name        = "AWSManagedRulesKnownBadInputsRuleSet"
+        vendor_name = "AWS"
+      }
+    }
+  }
+
+  lifecycle {
+    # Use `ignore_changes` to allow rotation without Terraform overwriting the value
+    ignore_changes = [rule]
+  }
+  tags = local.tags
+
+}
+
+# AWS Lambda Resources
+
+# IAM Role for Lambda Execution
+resource "aws_iam_role" "origin-secret-rotate-execution-role" {
+  name = "${var.application}-${var.environment}-origin-secret-rotate-role"
+
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [{
+      Effect    = "Allow"
+      Principal = { Service = "lambda.amazonaws.com" }
+      Action    = "sts:AssumeRole"
+    }]
+  })
+
+  inline_policy {
```

Comment:
This results in us getting a warning every time we run the terraform.

"The inline_policy argument is deprecated. Use the aws_iam_role_policy resource instead."

Reply:
Added data block and aws_iam_role_policy to attach to the role